source/userguide/dm.rst

```diff
@@ -567,6 +567,98 @@ There are three ways to exit a Dialogue Flow:
+
+This decorator replaces the need to define the ``@app.handle`` decorator. MindMeld will prompt the user for missing entities before applying the handler functionality by itself. The arguments required for this decorator are all the rules that would apply for that dialogue state (such as domain, intent, entities etc.), with the addition of a ``form`` argument.
+
+- ``form`` is a dictionary containing the following entries:
+
+  - ``entities`` (list, required): List of ``FormEntity`` objects with each defined for one entity slot to be filled. The order of entities provided in this list is important as the slots will be prompted in that order. So the order should follow the flow of the dialogue intended for slot filling.
+  - ``max_retries`` (int, optional, default 1): maximum number of retries allowed per entity or slot if user response is invalid.
+  - ``exit_msg`` (str, optional): If slot filling is exited abruptly without completion, define custom message to display.
+  - ``exit_keys`` (list, optional): List of exit hints for the slot filling flow. If these words or phrases are said by the user, the slot filling logic exits. Default: ['cancel', 'restart', 'exit', 'reset'].
+
+.. |br| raw:: html
+
+   <br />
+
+- ``FormEntity`` is a class that allows creation of entity objects for slot filling and comprises of the following attributes:
+  
+  - ``entity`` (str, required): Entity name
+  - ``role`` (str, optional): The role of the entity
+  - ``responses`` (list or str, optional): message for prompting the user for missing entities
+  - ``retry_response`` (str, optional): message for re-prompting users. If not provided, defaults to ``responses``.
+  - ``value`` (str, optional): The resolved value of the entity
+  - ``default_eval`` (bool, optional): Use system validation (default: True)
+  - ``hints`` (list, optional): Developer defined list of keywords to verify the user input against
+  - ``custom_eval`` (func, optional): custom validation function (should return bool:
+    validated or not)
+
+.. |br| raw:: html
+
+   <br />
+
+Once the slot filling is complete, the filled in entities can be access through ``request.entities`` in the same manner as any other handler.
+
+.. note::
+   |
+    The order of entities provided in the ``entities`` list in the form is important as the slots will be prompted in that order.
+
+Example use-case
+^^^^^^^^^^^^^^^^
+
+Transfer money in a banking app:
+
+.. code:: python
+
+    from mindmeld.core import FormEntity
+
+    form_transfermoney = {
+      'entities':[
+          FormEntity(
+              entity='account_en',
+              role='account_from',
+              responses=['Sure. Transfer from which account?'],
+              custom_eval=developer_func # validates the user-response for this entity 
```

Comment:
Maybe better to use custom_eval for 'sys_amount-of-money'. It could be something as simple as checking the "$" sign in the string.

```suggestion
              custom_eval=developer_func # validates the user-response for this entity 
```

Reply:
Added.